### PR TITLE
Add AllAnime

### LINKED ISSUE #14233: Source request for allanime.site/manga
### Source name

All Anime

### Source link

https://allanime.site/manga

### Language

English

### Other details

Good site with huge library and relatively unknown.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a title with source name.
- [X] I have checked that the extension does not already exist on the [website extensions list](https://tachiyomi.org/extensions/) or the app.
- [X] I have checked that the extension does not already exist by searching the [GitHub repository](https://github.com/tachiyomiorg/tachiyomi-extensions/) and verified it does not appear in the code base.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by mobi2002**
closes  #14233

Checklist:

- [ ] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

File: src/en/allanime/src/eu/kanade/tachiyomi/extension/en/allanime/AllAnime.kt [hunk 1]
```diff
@@ -0,0 +1,485 @@
...
+
+        SwitchPreferenceCompat(screen.context).apply {
+            key = SHOW_ADULT_PREF
+            title = "Show Adult Content"
+            setDefaultValue(false)
+
+            setOnPreferenceChangeListener { _, newValue ->
+                val showAdult = newValue as Boolean
+
+                preferences.edit().putBoolean(key, showAdult).commit()
```
Comment (stevenyomi) [inline src/en/allanime/src/eu/kanade/tachiyomi/extension/en/allanime/AllAnime.kt:467]:
```suggestion
                preferences.edit().putBoolean(key, showAdult).commit()
```

Remove all 3 lines of `commit()` as these are unnecessary.

Comment (stevenyomi) [reply src/en/allanime/src/eu/kanade/tachiyomi/extension/en/allanime/AllAnime.kt]: I mean removing them completely. The app handles saving the values already.

Comment (mobi2002) [reply src/en/allanime/src/eu/kanade/tachiyomi/extension/en/allanime/AllAnime.kt]: But the build fails without it.
```
Type mismatch: inferred type is Unit but Boolean was expected
```

File: src/en/allanime/src/eu/kanade/tachiyomi/extension/en/allanime/AllAnime.kt [hunk 2]
```diff
@@ -0,0 +1,460 @@
...
+
+    private val apiUrl = "https://api.$domain/allanimeapi"
+
+    override val client: OkHttpClient = network.cloudflareClient.newBuilder()
+        .rateLimitHost(apiUrl.toHttpUrl(), 1)
+        .build()
+
+    override fun headersBuilder() = Headers.Builder().apply {
+        add("Referer", "$baseUrl/")
+    }
```
Comment (stevenyomi) [inline src/en/allanime/src/eu/kanade/tachiyomi/extension/en/allanime/AllAnime.kt:69]:
```suggestion
    override fun headersBuilder() = super.headersBuilder()
        .add("Referer", "$baseUrl/")

```

Better use the one from superclass as it has default user agent. Not using is fine though as the app fallbacks to it using interceptor and other logic.

File: src/en/allanime/src/eu/kanade/tachiyomi/extension/en/allanime/AllAnime.kt [hunk 3]
```diff
@@ -0,0 +1,459 @@
...
+                put("page", page)
+                put("allowAdult", showAdult)
+                put("allowUnknown", false)
+            }
+            put("query", POPULAR_QUERY)
+        }
+
+        val body = payload.toString().toRequestBody(JSON_MEDIA_TYPE)
+
+        val newHeaders = headersBuilder()
```
Comment (arkon) [inline src/en/allanime/src/eu/kanade/tachiyomi/extension/en/allanime/AllAnime.kt:86]: Some stuff like how you're creating requests and parsing responses are kind of repetitive, you could consider refactoring it later.

**Pull request merged**